src/libstd/sys/windows/backtrace.rs

```diff
@@ -157,3 +160,37 @@ unsafe fn _write(w: &mut Write) -> io::Result<()> {
 
     Ok(())
 }
+
+fn query_full_process_image_name() -> io::Result<PathBuf> {
+    super::fill_utf16_buf(|buf, mut sz| unsafe {
+        c::QueryFullProcessImageNameW(c::GetCurrentProcess(), 0, buf, &mut sz); sz
```

Comment:
It looks like there's no need to convert this `Vec<u16>` vector into `PathBuf` and back, you can just use `Vec<u16>` through all the process.